src/cpp/utils/SystemInfo.hpp

```diff
@@ -71,6 +74,40 @@ class SystemInfo
         return singleton;
     }
 
+    /**
+     * Retrieve the value of a given environment variable if it exists, or nullptr.
+     * The c-string which is returned in the env_value output parameter is only valid until the next time this function
+     * is called, because it is a direct pointer to the static storage.
+     * Modifying the string returned in env_value invokes undefined behavior.
+     * If the environment variable is not set, a nullptr will be returned.
+     * 
+     * This function is thread-safe as long as no other function modifies the host environment (in particular, POSIX
+     * functions setenv, unsetenv and putenv would introduce a data race if called without synchronization.)
+     * 
+     * \param [in] env_name the name of the environment variable
+     * \param [out] env_value pointer to the value c-string
+     * @return RETCODE_OK if the environment variable is set.
+     * RETCODE_NO_DATA if the environment variable is unset.
+     * RETCODE_BAD_PARAMETER if the provided parameters are not valid.
+     */
+    static ReturnCode_t get_env(
+            const char* env_name, const char** env_value)
+    {
+        if (env_name == nullptr || env_value == nullptr)
```

Comment:
Should we consider an empty `env_name` as incorrect?

```suggestion
        if (env_name == nullptr || env_value == nullptr || *env_name == '\0')
```

Reply:
Done in 6c230616a2b92f483864849710847720b666c1c6